digiroad2-viite/src/main/scala/fi/liikennevirasto/viite/RoadAddressService.scala

```diff
@@ -613,22 +613,27 @@ class RoadAddressService(roadLinkService: RoadLinkService, eventbus: DigiroadEve
     Kalpa-API methods
   */
   def getRoadAddressesLinkByMunicipality(municipality: Int, roadLinkDataTempAPI:Boolean=false): Seq[RoadAddressLink] = {
-    //TODO: Remove null checks and make sure no nulls are generated
-    //TODO: All this request to VVH can be done using asynchronous
-    val roadLinks = {
-      val tempRoadLinks = roadLinkService.getViiteRoadLinksFromVVHByMunicipality(municipality,frozenTimeVVHAPIServiceEnabled)
-      if (tempRoadLinks == null)
-        Seq.empty[RoadLink]
-      else tempRoadLinks
-    }
-    val complimentaryLinks = {
-      val tempComplimentary = roadLinkService.getComplementaryRoadLinksFromVVH(municipality)
-      if (tempComplimentary == null)
-        Seq.empty[RoadLink]
-      else tempComplimentary
-    }
+
+    val (roadLinksWithComplementary, _) =
```

Comment:
do we really need two val definition here?

Reply:
I do believe so if you check bellow on the allRoadLinks (line 637) it is using the roadLinksWithComplementary value

Reply:
yes we need that value, the left one from the definition. the , _ can be ignored if we change
"(roadLinks ++ complimentaryLinks, Seq())" to (roadLinks ++ complimentaryLinks)
and roadLinkService.getRoadLinksWithComplementaryAndChangesFromVVH(municipality) to roadLinkService.getRoadLinksWithComplementaryAndChangesFromVVH(municipality)._1

Reply:
It is using the two val because the method that gets the cached road links also gets the changed. And so that it also works with the frozen i just added the Seq(). Since the frozen if statement part will be removed soon.